app/src/main/java/com/example/playlistmaker/SearchActivity.kt

```diff
@@ -65,38 +103,64 @@ class SearchActivity : AppCompatActivity() {
     }
 
     private fun clearButtonVisibility(s: CharSequence?): Int {
+        showMessage("",null, false)
         return if (s.isNullOrEmpty()) {
             View.GONE
         } else {
             View.VISIBLE
         }
     }
-}
 
+    private fun updateListTracks(){
+
+        if(!savedSearchEditText.isNullOrEmpty()){
+            playlistRetrofit.search(savedSearchEditText!!)
```

Comment:
🍏 Лучше избегать использования оператора !!. Особенно с изменяемыми переменными. Теоретически к моменту использования в строке 117 переменная могла уже измениться и стать null.
Можно сделать так:
```suggestion
 val text = savedSearchEditText
 if(!text.isNullOrEmpty()){
            playlistRetrofit.search(text) 
            ...
            }
```